sklearn/metrics/tests/test_classification.py

```diff
@@ -408,17 +408,27 @@ def test_matthews_corrcoef():
     y_true_inv2 = np.where(y_true_inv2, 'a', 'b')
     assert_almost_equal(matthews_corrcoef(y_true, y_true_inv2), -1)
 
+    # Some architectures do not provide warnings on divide by zero
+    clean_warning_registry()
+    with warnings.catch_warnings(record=True) as w:
+        1. / np.array([0.])
+        numpy_provides_div0_warning = len(w) == 1
```

Comment:
Side comment: I imagine it could have been worth exposing the `numpy_provides_div0_warning` under some module in `sklearn.utils` if it was re-used elsewhere, but that doesn't seem to be the case now. Essentially it's a platform compatibility flag in line with other fixes in `sklearn/utils/fixes.py`.